src/utils/contractToRead.ts

```diff
@@ -0,0 +1,14 @@
+import { Contract } from '@ethersproject/contracts'
+
+type ContractConfig<T> = T | Contract | undefined
+
+export function contractToRead<T extends string>(
```

Comment:
since this util is outside the scope of actually reading the contract, maybe name it something like "getContract"?

Reply:
Solid - totally agree. `getContract` is a much better name